src/main/org/codehaus/groovy/classgen/AsmClassGenerator.java

```diff
@@ -1251,11 +1251,13 @@ private void processClassVariable(String name) {
             mv.visitInsn(DUP);
 
             loadThisOrOwner();
-            mv.visitLdcInsn(name);
+            mv.visitLdcInsn(expression.getName());
 
             mv.visitMethodInsn(INVOKESPECIAL, "org/codehaus/groovy/runtime/ScriptReference", "<init>", "(Lgroovy/lang/Script;Ljava/lang/String;)V", false);
         } else {
-            PropertyExpression pexp = new PropertyExpression(new VariableExpression("this"), name);
+            PropertyExpression pexp = new PropertyExpression(new VariableExpression("this"), expression.getName());
```

Comment:
The patch provided in the JIRA creates a new `VariableExpression` (see below), I think that needs to be done so the source position is not set on the constant `VariableExpress.THIS_EXPRESSION` via the `getObjectExpression().setSourcePosition(..)` call.

```java
PropertyExpression pexp = new PropertyExpression(new VariableExpression("this", ClassHelper.DYNAMIC_TYPE), name);
```